force-app/main/default/classes/JoobleSaveToSalesforce.cls

```diff
@@ -0,0 +1,114 @@
+    
+            for(Object record : jobRecords) {
+                Map<String, Object> dataObject = (Map<String,Object>) record;
+
+                Job_Application__c job = new Job_Application__c();
+                Long joobleId = (Long) dataObject.get('id');
+                String joobleIdString = String.valueOf(joobleId);
+                String companyName = (String) dataObject.get('company');
+                job.Name = (String) dataObject.get('title');
+                job.Location__c = (String) dataObject.get('location');
+                job.Position_Title__c = (String) dataObject.get('title');
+                String salaryString = (String) dataObject.get('salary');
+                job.Salary__c = String.isEmpty(salaryString) ? 0 : Decimal.valueOf(salaryString);
+                job.Description__c = (String) dataObject.get('snippet');
+                job.URL__c = (String) dataObject.get('link');
+                String source = 'Source: ' + (String) dataObject.get('source');
+                job.Notes__c = source;
+                job.Status__c = 'Saved';
+    
+                mapJoobleIdToJobApplication.put(joobleIdString, job);
+                mapJoobleIdToCompanyName.put(joobleIdString, companyName);
+            }
+
+            // Process Company Names 
+            Map<String, Account> mapAccountNameToAccount = processCompanyNames(mapJoobleIdToCompanyName);
+
+            // Create Map of JoobleId : Accounts 
+            Map<String, Account> mapJoobleIdToAccount = new Map<String, Account>();
+            for (String joobleIdString : mapJoobleIdToCompanyName.keySet()) {
+                mapJoobleIdToAccount.put(joobleIdString, mapAccountNameToAccount.get(mapJoobleIdToCompanyName.get(joobleIdString)));
+            }
+
+            // Update Map of Jooble Id : Applications with Account Id from mapJoobleIdToAccount
+            for(String joobleIdString : mapJoobleIdToJobApplication.keySet()) {
+                mapJoobleIdToJobApplication.get(joobleIdString).Company_Name__c = mapJoobleIdToAccount.get(joobleIdString).Id;
+            }
+
+            // Insert Job Application records
+            insert mapJoobleIdToJobApplication.values();
+
+        } catch (Exception e) {
+           ErrorLogUtility.logError(e);
+
+        }
+       
+    }
+
+    // Process Company Names and insert to Salesforce if needed
+    private static Map<String, Account> processCompanyNames(Map<String, String> mapJoobleIdToCompanyName){
+
+
+        // Create Map of Account Name : Account
+        Map<String, Account> mapAccountNameToAccount = new Map<String, Account>();
+
+        try{
+    
+            Set<String> companyNames = new Set<String>(mapJoobleIdToCompanyName.values());  
+
+            // Get existing accounts in Salesforce that match the Company Names
+            List<Account> existingAccounts = [ 
+                SELECT Id, Name
+                FROM Account
+                WHERE Name IN :companyNames
```

Comment:
Did you encounter any issues with duplicate Accounts getting created due to data inconsistencies on Jooble's end? i.e. some jobs listing company as 'Cloud Code Academy' and other positions with the company as 'CCA' or 'Cloud Code'